### PR TITLE
(IAC-716) Add troubleshooting documentation

### PR DESCRIPTION
**Pull request by michaeltlombardi**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -73,15 +52,87 @@ Verify the resources are created successfully.
...
-### Platform Support
+## Troubleshooting
 
-- Windows PowerShell 5.1
+Occasionally, something will go wrong when trying to apply a manifest containing Puppetized DSC resources.
+There's a few different ways this can happen:
+
+- There can be a misgenerated type, in which case the Puppet representation of the DSC resource does not match the resource's API surface.
+  This can be mismatched enums or types - in these cases, the issue is with this builder and a [bug will need to be submitted for this project](https://github.com/puppetlabs/PuppetDscBuilder/issues/new?assignees=&labels=Bug%2C+bug&template=bug-report.md&title=) so the Puppetized module can be rebuilt with the patched builder.
+- There can be an error in the ruby code that translates the DSC resource back and forth between Puppet and PowerShell, in which case the issue is with the base provider that lives in the [`pwshlib` module](https://forge.puppet.com/puppetlabs/pwshlib);
```
Comment (sanfrancrisko) [inline README.md:62]:
```suggestion
- There can be an error in the Ruby code that translates the DSC resource back and forth between Puppet and PowerShell, in which case the issue is with the base provider that lives in the [`pwshlib` module](https://forge.puppet.com/puppetlabs/pwshlib);
```

File: README.md [hunk 2]
```diff
@@ -73,15 +52,87 @@ Verify the resources are created successfully.
...
+Debug: dsc_psrepository: Returned to Puppet as {:dsc_sourcelocation=>"https://www.powershellgallery.com/api/v2", :dsc_installationpolicy=>"Trusted", :dsc_name=>"psgAllery", :dsc_packagemanagementprovider=>"NuGet", :dsc_publishlocation=>"https://www.powershellgallery.com/api/v2/package/", :dsc_ensure=>"Present", :dsc_scriptpublishlocation=>"https://www.powershellgallery.com/api/v2/package/", :dsc_scriptsourcelocation=>"https://www.powershellgallery.com/api/v2/items/psscript", :name=>"PowerShell Gallery", :ensure=>"present"}
+Debug: dsc_psrepository: Canonicalized Resources: [{:dsc_installationpolicy=>"Trusted", :dsc_name=>"psgAllery", :name=>"PowerShell Gallery"}]
+... (snipped for brevity)
+```
+
+The debug information tells you what Puppet is up to, elaborates on the values being passed, then gives you the full text of the PowerShell script being executed (with any sensitive info redacted), and finally tells you what the outcome of that run was.
```
Comment (sanfrancrisko) [inline README.md:129]:
```suggestion
The debug information tells you what Puppet is up to, elaborates on the values being passed, then gives you the full text of the PowerShell script being executed (with any sensitive info redacted). Finally, it tells you what the outcome of that run was.
```

**Review by sanfrancrisko — APPROVED** [review]
Reads well! 👍  

Just a few minor suggestions but happy for them to be dismissed - all quite subjective.

**Comment by michaeltlombardi** [comment]
@sanfrancrisko thanks! Updated based on your feedback! 😊

**Review by sanfrancrisko — APPROVED** [review]

**Pull request merged**